torchsde/_core/adjoint_sde.py

```diff
@@ -0,0 +1,425 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import torch
+
+from . import base_sde
+from . import misc
+from ..settings import SDE_TYPES, NOISE_TYPES
+
+
+class AdjointSDE(base_sde.BaseSDE):
+
+    def __init__(self, forward_sde, params, n_tensors, logqp=False):
+        # There's a mapping from the noise type of the forward SDE to the noise type of the adjoint.
+        # Usually, these two aren't the same, e.g. when the forward SDE has additive noise, the adjoint SDE's diffusion
+        # is a linear function of the adjoint variable, so it is not of additive noise.
+        sde_type = forward_sde.sde_type
+        if sde_type == SDE_TYPES.ito:
```

Comment:
The Ito/Strat check isn't necessary here, the same thing happens on both branches.